src/main/java/seedu/address/logic/commands/CountryNoteEditCommand.java

```diff
@@ -0,0 +1,120 @@
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.note.CountryNote;
+import seedu.address.model.tag.Tag;
+import seedu.address.ui.WidgetViewOption;
+
+/**
+ * A class that encapsulates the logic for editing country notes.
+ */
+public class CountryNoteEditCommand extends Command {
+
+    public static final String COMMAND_WORD = "country note edit";
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Edits the country note at the given index in the last viewed country note list panel.\n"
+            + "Parameters: INDEX "
+            + "(" + PREFIX_NOTE + "NOTE_STRING )"
+            + " (" + PREFIX_TAG + "TAG)...\n"
+            + "Example: " + COMMAND_WORD + " 1 " + PREFIX_NOTE + "better government stability in recent months";
+    public static final String MESSAGE_SUCCESS = "Edited country note at index %1$s: %2$s";
+    private final Index targetIndex;
+    private final CountryNote countryNote;
+    private final Set<Tag> tags;
+
+    /**
+     * Initializes a CountryNoteEditCommand with the given targetIndex.
+     *
+     * @param targetIndex The given targetIndex.
+     * @param countryNote The country note that contains the new note content.
+     */
+    public CountryNoteEditCommand(Index targetIndex, CountryNote countryNote) {
+        requireAllNonNull(targetIndex);
+        this.targetIndex = targetIndex;
+        this.countryNote = countryNote;
+        this.tags = new HashSet<>();
+    }
+
+    /**
+     * Initializes a CountryNoteEditCommand with the given targetIndex.
+     *
+     * @param targetIndex The given targetIndex.
+     * @param tags The new tags to add to the country note at the existing targetIndex.
+     */
+    public CountryNoteEditCommand(Index targetIndex, Set<Tag> tags) {
+        requireAllNonNull(targetIndex, tags);
+        this.targetIndex = targetIndex;
+        this.countryNote = CountryNote.NULL_COUNTRY_NOTE;
+        this.tags = tags;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<CountryNote> lastShownList = model.getFilteredCountryNoteList();
+
+        if (targetIndex.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_COUNTRY_NOTE_DISPLAYED_INDEX);
+        }
+
+        CountryNote countryNoteToEdit = lastShownList.get(targetIndex.getZeroBased());
+        CountryNote newCountryNote;
+        tags.addAll(countryNoteToEdit.getTags());
+
+        if (countryNote.equals(CountryNote.NULL_COUNTRY_NOTE)) {
+            newCountryNote = new CountryNote(countryNoteToEdit);
+        } else {
+            newCountryNote = countryNote.setCountry(countryNoteToEdit.getCountry());
+            tags.addAll(countryNote.getTags());
+        }
+
+        if (tags.size() > 1 && tags.contains(Tag.UNTAGGED)) {
+            tags.remove(Tag.UNTAGGED);
+        }
+
```

Comment:
Need some comments here, quite hard to understand what this code is doing.  

Imo I think the constructor `CountryNote(CountryNote note)`  and the method `setCountry` is not needed, instead we can just have a CountryNote constructor with content, country and tags, and then port the countryNoteToEdit fields over as needed. 

Feels like that'll make the code much clearer.

Also, I think .remove for sets won't throw any exception even if that item isn't inside the set, so its safe to remove the contains check

Reply:
Created new constructor with content, country and tags and added comments for this part